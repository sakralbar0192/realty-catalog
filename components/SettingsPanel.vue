<template>
  <Teleport to="body">
    <Transition name="slide-fade">
      <div
        v-if="isOpen"
        :class="styles.overlay"
        @click="close"
        @keydown.escape="close"
        tabindex="-1"
        data-test="settings-overlay"
      >
        <div
          :class="styles.panel"
          @click.stop
          data-test="settings-panel"
        >
          <div :class="styles.header">
            <h3 :class="styles.title">{{ $t('settings.title') }}</h3>
            <button
              :class="styles.closeButton"
              @click="close"
              :aria-label="$t('common.close')"
              data-test="settings-close"
            >
              <Icon name="x" :size="20" />
            </button>
          </div>

          <div :class="styles.content">
            <div :class="styles.section">
              <h4 :class="styles.sectionTitle">{{ $t('theme.title') }}</h4>
              <ThemeToggle />
            </div>

            <div :class="styles.section">
              <h4 :class="styles.sectionTitle">{{ $t('common.language') }}</h4>
              <LanguageSwitcher />
            </div>
          </div>
        </div>
      </div>
    </Transition>
  </Teleport>
</template>

<script setup lang="ts">
import { ref } from 'vue'
import Icon from '~/components/Icon.vue'
import ThemeToggle from '~/components/ThemeToggle.vue'
import LanguageSwitcher from '~/components/LanguageSwitcher.vue'
import { useAppI18n } from '~/composables/useI18n'
import styles from '~/assets/styles/components/SettingsPanel.module.scss'

const { translate: $t } = useAppI18n()

const isOpen = ref(false)

const open = () => {
  isOpen.value = true
}

const close = () => {
  isOpen.value = false
}

// Expose methods for parent components
defineExpose({
  open,
  close,
  isOpen
})
</script>