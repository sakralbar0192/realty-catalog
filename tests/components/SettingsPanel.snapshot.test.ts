import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import SettingsPanel from '../../components/SettingsPanel.vue'

// Mock the useAppI18n composable
vi.mock('../../composables/useI18n', () => ({
  useAppI18n: () => ({
    translate: (key: string) => {
      const translations: Record<string, string> = {
        'settings.title': 'Settings',
        'common.close': 'Close',
        'theme.title': 'Theme',
        'common.language': 'Language',
      }
      return translations[key] || key
    },
  }),
}))

describe('SettingsPanel Snapshots', () => {
  it('should match snapshot when closed', () => {
    const wrapper = mount(SettingsPanel, {
      global: {
        stubs: {
          LanguageSwitcher: {
            name: 'LanguageSwitcher',
            template: '<div>Language Switcher</div>',
          },
          ThemeToggle: {
            name: 'ThemeToggle',
            template: '<div>Theme Toggle</div>',
          },
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot when open', async() => {
    const wrapper = mount(SettingsPanel, {
      attachTo: document.body,
      global: {
        stubs: {
          LanguageSwitcher: {
            name: 'LanguageSwitcher',
            template: '<div>Language Switcher</div>',
          },
          ThemeToggle: {
            name: 'ThemeToggle',
            template: '<div>Theme Toggle</div>',
          },
        },
      },
    })

    await wrapper.vm.open()
    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })
})
