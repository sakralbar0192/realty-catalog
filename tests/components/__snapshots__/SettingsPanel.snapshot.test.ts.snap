// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettingsPanel Snapshots > should match snapshot when closed 1`] = `
"<!--teleport start-->
<!--teleport end-->"
`;

exports[`SettingsPanel Snapshots > should match snapshot when open 1`] = `
"<!--teleport start-->
<!--teleport end-->"
`;
